### PR TITLE
Add support for multiple CMA transfers when size > max supported

### PR DESCRIPTION
**Pull request by nitbhat**

### CODE REVIEW

File: tests/charm++/zerocopy/Makefile [hunk 1]
```diff
@@ -6,9 +6,15 @@ DIRS = \
...
   bcast_nonzero_root \
   pup_buffer \
+  large_p2p \
+
+NONSCALEDIRS = \
+  large_p2p \
 
 TESTDIRS = $(DIRS)
 
+TESTPDIRS = $(filter-out $(NONSCALEDIRS),$(TESTDIRS))
```
Comment (evan-charmworks) [inline tests/charm++/zerocopy/Makefile:16]: The `testp` rule below should have `TESTDIRS` changed to `TESTPDIRS`.

File: src/arch/util/machine-cma.C [hunk 2]
```diff
@@ -3,6 +3,8 @@
 #endif
 #include <sys/uio.h>
 
+#define MAX_CMA_RW_COUNT 2147479552
```
Comment (evan-charmworks) [inline src/arch/util/machine-cma.C:6]: Maybe this should be the hexadecimal literal `0x7ffff000` for clarity.

File: tests/charm++/zerocopy/large_p2p/Makefile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+large_p2p.o: large_p2p.C cifiles
+	$(CHARMC) -c large_p2p.C
+
+test: all
+	$(call run, +p1 ./large_p2p 2147479558 )
+	$(call run, +p2 ./large_p2p 2147479558 )
+ifeq ($(CMK_USE_CMA),1)
+	$(call run, +p2 ./large_p2p 2147479558 +noCMAForZC )
+endif
```
Comment (evan-charmworks) [inline tests/charm++/zerocopy/large_p2p/Makefile:24]: Does this test succeed on i386?

File: tests/charm++/zerocopy/Makefile [hunk 4]
```diff
@@ -6,9 +6,15 @@ DIRS = \
   dereg_and_nodereg \
```
Comment (evan-charmworks) [inline tests/charm++/zerocopy/Makefile:6]: The `largedata` line here should be changed to `large_bcast` to match the directory rename.

File: tests/charm++/zerocopy/large_p2p/Makefile (hunk 3 above)
Comment (evan-charmworks) [reply tests/charm++/zerocopy/large_p2p/Makefile:24]: Could also add a comment saying that the parameter passed in is `MAX_CMA_RW_COUNT + 6`.

File: tests/charm++/zerocopy/Makefile (hunk 4 above)
Comment (nitbhat) [reply tests/charm++/zerocopy/Makefile:6]: Done.

File: src/arch/util/machine-cma.C (hunk 2 above)
Comment (nitbhat) [reply src/arch/util/machine-cma.C]: Done.

File: tests/charm++/zerocopy/Makefile (hunk 1 above)
Comment (nitbhat) [reply tests/charm++/zerocopy/Makefile:16]: Done.

File: tests/charm++/zerocopy/large_p2p/Makefile (hunk 3 above)
Comment (nitbhat) [reply tests/charm++/zerocopy/large_p2p/Makefile:24]: Yes, it ran fine on `netlrts-linux-i386` build. Is that what you were asking?

Comment (nitbhat) [reply tests/charm++/zerocopy/large_p2p/Makefile:24]: Added the comment.

Comment (evan-charmworks) [reply tests/charm++/zerocopy/large_p2p/Makefile:24]: > Yes, it ran fine on `netlrts-linux-i386` build. Is that what you were asking?

Yes.

**Comment by nitbhat** [comment]
It looks like our test environment will not support the large_p2p test because of its memory requirements. 
```

../../../../bin/testrun  +p2 ./large_p2p 2147479558  ++local
Charmrun> scalable start enabled. 
Charmrun> started all node programs in 0.007 seconds.
Charm++> Running in non-SMP mode: 2 processes (PEs)
Converse/Charm++ Commit ID: v6.11.0-devel-351-g09fb0720e
Charm++ built with internal error checking enabled.
Do not use for performance benchmarking (build without --enable-error-checking to do so).
Isomalloc> Synchronized global address space.
Charm++> scheduler running in netpoll mode.
CharmLB> Load balancer assumes all CPUs are same.
Charm++> Running on 1 hosts (1 sockets x 1 cores x 2 PUs = 2-way SMP)
Charm++> cpu topology info is gathered in 0.002 seconds.
[0][0][0] Large Get Done
------------- Processor 0 Exiting: Called CmiAbort ------------
Reason: Could not malloc()--are we out of memory? (used: 18446744071562.066MB)
```

**Pull request merged**